agent/stats/engine.go

```diff
@@ -793,3 +809,57 @@ func (engine *DockerStatsEngine) ContainerDockerStats(taskARN string, containerI
 
 	return containerStats, containerNetworkRateStats, nil
 }
+
+// getContainerStatusMessage reads container's status message file, updates dockerIdToStatusMessageSince mapping
+// and returns the file contents when
+//  1. status message file for a firelens v2 container is being read for the first time.
+//  2. contents of status message file has been modified since last published to TACS.
+//  3. unable retrieve modified time for status message file.
+// getContainerStatusMessage returns empty string for statusMessage when
+//  1. status message file does not exist for firelens v2 container.
+//  2. error while reading the status message file.
+// getContainerStatusMessage returns true to indicate statusMessage is Nil to TACS when
+//  1. when a container is not a firelens v2 container.
+//  2. error while retrieving StatusMessageReportingPath.
+//  3. error reading existing status message file.
+//  4. contents of status message file has not been modified since last published to TACS.
+func (engine *DockerStatsEngine) getContainerStatusMessage(dockerContainer *apicontainer.DockerContainer, taskARN string) (string, bool) {
+	firelensConfig := dockerContainer.Container.GetFirelensConfig()
+	if firelensConfig == nil || (firelensConfig != nil && firelensConfig.Version != "v2") {
+		return "", true
+	}
+
+	StatusMessageFilepath, err := getStatusMessageReportingPath(engine.config.DataDirOnHost, taskARN, dockerContainer.Container.Name)
+	if err != nil {
+		seelog.Errorf("stats engine: %v", err)
```

Comment:
nit: more meaningful log, e.g. "failed to get status message reporting path: %v"